packages/vite/src/node/plugins/css.ts

```diff
@@ -828,6 +895,10 @@ async function compileCSS(
   modules?: Record<string, string>
   deps?: Set<string>
 }> {
+  if (config.css?.transformer === 'LightningCSS') {
+    return compileLightningCSS(id, code, config, config.css, urlReplacer)
```

Comment:
Should it be possible to use both Lightning CSS and PostCSS/SASS/Less together? In my prototype, I ran those pre-processors first (if needed), and then fed the output into Lightning CSS. That would make it possible to use things like Tailwind, but still use Lightning CSS for other transforms instead of autoprefixer for example.

Reply:
This can be done in build to get better minification. I'm not sure of the benefit in dev (compare to the complexity of not messing up with sourcemaps for example)

Reply:
Lightning CSS accepts an input source map so you can pass the source map from postcss/sass in and it'll remap the output. Aside from that, is there any reason _not_ to support it? Lightning CSS has different ways of compiling modern CSS features for older browsers than postcss-preset-env/autoprefixer, for example, and even includes support for some features that other compilers don't. Until recently, Lightning CSS was the only tool supporting relative color syntax, for example. Users may also want to use custom Lightning CSS transform plugins.

I think limiting it to be _only_ Lightning CSS _or_ PostCSS/Sass will severely limit the adoption, since so many people use Tailwind. I think it will be useful to give users the choice of what combination of tools they want to use.

Reply:
Yeah relative color syntax is a good example.

The main reason for not adding it for now was to reduce the API surface and so the number of bugs that then we have to consider. On top of that the API would not be as trivial (for example on how to be explicit on CSS modules being handled by one or the other). And in the future we will able to use Tailwind without PostCSS.

What's your opinion @patak-dev?

Reply:
I'm also leaning toward starting with Lightning CSS without preprocessing. But I think it is a case we should think about up front and design the config API to allow us to go there later. Using postcss as a preprocessor for lightning CSS, we could do it as:
```
css: {
  transformer: 'lightingcss',
  lightningcss: { ... },
  postcss: { ... }
}
```
Or maybe allow postcss options as `preprocessorOptions.postcss`?

Reply:
For now we are on the side of not being able to use pre-processor & LightningCSS. The elephant in the room is Tailwind, but given the fact that Tailwind without postcss is something coming and UnoCSS can give you the same experience without the PostCSS requirement (for people like liking the config part with fixed number of possibilities, I built [this](https://github.com/ArnaudBarre/downwind))